### PR TITLE
Introspection tests

### PR DESCRIPTION
**Pull request by jelly**

### CODE REVIEW

File: systemd_ctypes/introspection.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 def parse_method(method):
     return {
-        "in": [tag.attrib['type'] for tag in method.findall("arg[@direction!='out']")],
+        "in": [tag.attrib['type'] for tag in method.findall("arg[@direction='in']")],  # != is Python 3.10 only
```
Comment (allisonkarlitskaya) [inline systemd_ctypes/introspection.py:22]: This change is a no-go.  The spec states that if the direction is left out, it needs to be interpreted as `in`.

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]
I'm good with this in theory, but the change to the parsing of in arguments is not OK.  In fact, I'd prefer to see the addition of a unit test to confirm this exact behaviour (ie: argument specified without a direction is taken to be an "in")

File: systemd_ctypes/introspection.py (hunk 1 above)
Comment (jelly) [reply systemd_ctypes/introspection.py]: Ah, good point, bit inelegant fix but seems to work.

File: systemd_ctypes/introspection.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 def parse_method(method):
     return {
-        "in": [tag.attrib['type'] for tag in method.findall("arg[@direction!='out']")],
```
Comment (allisonkarlitskaya) [inline systemd_ctypes/introspection.py:22]: This is a behaviour change, but imo a good one.

**Review by allisonkarlitskaya — APPROVED** [review]
Thanks!

**Pull request merged**